apps/designer/app/designer/features/props-panel/use-props-logic.ts

```diff
@@ -127,9 +125,39 @@ export const usePropsLogic = ({
       const isPropRequired = draft[index].required;
 
       if (isPropRequired !== true) {
-        draft[index].prop = name;
-        draft[index].value = defaultValue;
+        // @todo we need to now allow to change non required on required prop too
+        draft[index] = {
+          id: draft[index].id,
+          prop: name,
+          value: defaultValue,
+        };
+      }
+    })(userProps);
+
+    // Optimistically update the state (what if publish fails?)
+    setUserProps(nextUserProps);
+
+    const updatedProps = nextUserProps.filter((item) => item.id === id);
+
+    updateProps(updatedProps);
+  };
+
+  const handleChangePropValue: HandleChangePropValue = (id, value) => {
+    const nextUserProps = produce((draft: Array<UserProp>) => {
+      const index = draft.findIndex((item) => item.id === id);
+      const val = draft[index];
+
+      if ("value" in val) {
+        val.value = value;
+        return;
       }
+
+      draft[index] = {
+        id: val.id,
+        prop: val.prop,
+        required: val.required,
+        value,
+      };
     })(userProps);
```

Comment:
I'd move logic like this into `/shared` somewhere. Similar to this https://github.com/webstudio-is/webstudio-designer/blob/55280b8348a8a12cdabdd15eced6fa0aa2223a0c/apps/designer/app/shared/props-utils/delete-prop.ts#L3-L13

Reply:
I'm not getting what's shared here. And why to move handlers into shared

Reply:
I meant we could share something like `setPropValueMutable(allProps: AllUserProps, name: ..., value: ...): void`. But after you've updated the code maybe it doesn't make as much sense as the code became trivial